### ISSUE TITLE
crash when removing device from JBOD

### ISSUE THREAD

**Issue by mabod** [openzfs/zfs#6656]
### System information
```
18# inxi 
CPU~Quad core Intel Core i7-7700K (-HT-MCP-) speed/max~4200/4500 MHz Kernel~4.13.2-1-MANJARO x86_64 Up~4 min Mem~2563.1/16010.8MB HDD~19323.9GB(3.3% used) Procs~485 Client~Shell inxi~2.3.38  
```
 
modinfo zfs | grep -iw version:
version:        0.7.0-1

modinfo spl | grep -iw version:
version:        0.7.0-1

### Describe the problem you're observing

zfs crashing. zfs hangs. Only reboot helps. 

### Describe how to reproduce the problem

I have a JBOD in a fantec usb case. 3 drives making up one raidz. I use this for testing purposes. 

```
21# zpool status zf2-raidz
  pool: zf2-raidz
 state: ONLINE
  scan: scrub repaired 0B in 0h8m with 0 errors on Tue Sep 19 07:58:49 2017
config:

	NAME                         STATE     READ WRITE CKSUM
	zf2-raidz                    ONLINE       0     0     0
	  raidz1-0                   ONLINE       0     0     0
	    fantec2-Seagate-500SSHD  ONLINE       0     0     0
	    fantec2-WD750            ONLINE       0     0     0
	    fantec2-WD500            ONLINE       0     0     0
```
I wanted to test the drive fail behaviour of zfs. I started a dd file copy and in between I unplugged one drive form the raidz. the dd came to a halt and the following crash is reported in the journal:

```
Sep 19 07:40:07 rakete kernel: INFO: task txg_sync:29104 blocked for more than 120 seconds.
Sep 19 07:40:07 rakete kernel:       Tainted: P           O    4.13.2-1-MANJARO #1
Sep 19 07:40:07 rakete kernel: "echo 0 > /proc/sys/kernel/hung_task_timeout_secs" disables this message
Sep 19 07:40:07 rakete kernel: txg_sync        D    0 29104      2 0x00000000
Sep 19 07:40:07 rakete kernel: Call Trace:
Sep 19 07:40:07 rakete kernel:  __schedule+0x239/0x890
Sep 19 07:40:07 rakete kernel:  schedule+0x3d/0x90
Sep 19 07:40:07 rakete kernel:  io_schedule+0x16/0x40
Sep 19 07:40:07 rakete kernel:  cv_wait_common+0xb0/0x130 [spl]
Sep 19 07:40:07 rakete kernel:  ? wait_woken+0x80/0x80
Sep 19 07:40:07 rakete kernel:  __cv_wait_io+0x18/0x20 [spl]
Sep 19 07:40:07 rakete kernel:  zio_wait+0xf2/0x1b0 [zfs]
Sep 19 07:40:07 rakete kernel:  dsl_pool_sync+0xb8/0x440 [zfs]
Sep 19 07:40:07 rakete kernel:  spa_sync+0x43f/0xd80 [zfs]
Sep 19 07:40:07 rakete kernel:  txg_sync_thread+0x2d2/0x4a0 [zfs]
Sep 19 07:40:07 rakete kernel:  ? finish_task_switch+0x75/0x200
Sep 19 07:40:07 rakete kernel:  ? txg_delay+0x170/0x170 [zfs]
Sep 19 07:40:07 rakete kernel:  thread_generic_wrapper+0x72/0x80 [spl]
Sep 19 07:40:07 rakete kernel:  kthread+0x125/0x140
Sep 19 07:40:07 rakete kernel:  ? __thread_exit+0x20/0x20 [spl]
Sep 19 07:40:07 rakete kernel:  ? kthread_create_on_node+0x70/0x70
Sep 19 07:40:07 rakete kernel:  ret_from_fork+0x25/0x30
```
zfs is telling me that a device is missing and I need to plug it in again and do a "zpool clear" to recover. I tried that but it did not help. zfs did not recover. In fact the issue is, that the fantec hardware is resetting the complete usb bus if one device is plugged in or unplugged. So the hardware is not hotplug ready.

When I plugin the drive again, the usb bus is resetted, my systemd automount service kicks in and tries to mount the pool "zf2-raidz" again. 

The automount servcie works fine. It looks like this:

```
23# cat /etc/systemd/system/zf2-raidz-automount.service
[Unit]
After=dev-disk-by\x2dvdev-fantec2\x2dWD750\x2dpart1.device

[Service]
ExecStart=/usr/sbin/zpool import zf2-raidz
ExecStartPost=/usr/bin/logger "started ZFS pool zf2-raidz"

[Install]
WantedBy=dev-disk-by\x2dvdev-fantec2\x2dWD750\x2dpart1.device
```

The crash report is this:

```
Sep 19 07:42:10 rakete kernel: INFO: task txg_sync:29104 blocked for more than 120 seconds.
Sep 19 07:42:10 rakete kernel:       Tainted: P           O    4.13.2-1-MANJARO #1
Sep 19 07:42:10 rakete kernel: "echo 0 > /proc/sys/kernel/hung_task_timeout_secs" disables this message
Sep 19 07:42:10 rakete kernel: txg_sync        D    0 29104      2 0x00000000
Sep 19 07:42:10 rakete kernel: Call Trace:
Sep 19 07:42:10 rakete kernel:  __schedule+0x239/0x890
Sep 19 07:42:10 rakete kernel:  schedule+0x3d/0x90
Sep 19 07:42:10 rakete kernel:  io_schedule+0x16/0x40
Sep 19 07:42:10 rakete kernel:  cv_wait_common+0xb0/0x130 [spl]
Sep 19 07:42:10 rakete kernel:  ? wait_woken+0x80/0x80
Sep 19 07:42:10 rakete kernel:  __cv_wait_io+0x18/0x20 [spl]
Sep 19 07:42:10 rakete kernel:  zio_wait+0xf2/0x1b0 [zfs]
Sep 19 07:42:10 rakete kernel:  dsl_pool_sync+0xb8/0x440 [zfs]
Sep 19 07:42:10 rakete kernel:  spa_sync+0x43f/0xd80 [zfs]
Sep 19 07:42:10 rakete kernel:  txg_sync_thread+0x2d2/0x4a0 [zfs]
Sep 19 07:42:10 rakete kernel:  ? finish_task_switch+0x75/0x200
Sep 19 07:42:10 rakete kernel:  ? txg_delay+0x170/0x170 [zfs]
Sep 19 07:42:10 rakete kernel:  thread_generic_wrapper+0x72/0x80 [spl]
Sep 19 07:42:10 rakete kernel:  kthread+0x125/0x140
Sep 19 07:42:10 rakete kernel:  ? __thread_exit+0x20/0x20 [spl]
Sep 19 07:42:10 rakete kernel:  ? kthread_create_on_node+0x70/0x70
Sep 19 07:42:10 rakete kernel:  ret_from_fork+0x25/0x30
Sep 19 07:42:10 rakete kernel: INFO: task dd:29417 blocked for more than 120 seconds.
Sep 19 07:42:10 rakete kernel:       Tainted: P           O    4.13.2-1-MANJARO #1
Sep 19 07:42:10 rakete kernel: "echo 0 > /proc/sys/kernel/hung_task_timeout_secs" disables this message
Sep 19 07:42:10 rakete kernel: dd              D    0 29417  27419 0x00000004
Sep 19 07:42:10 rakete kernel: Call Trace:
Sep 19 07:42:10 rakete kernel:  __schedule+0x239/0x890
Sep 19 07:42:10 rakete kernel:  schedule+0x3d/0x90
Sep 19 07:42:10 rakete kernel:  cv_wait_common+0x11c/0x130 [spl]
Sep 19 07:42:10 rakete kernel:  ? wait_woken+0x80/0x80
Sep 19 07:42:10 rakete kernel:  __cv_wait+0x15/0x20 [spl]
Sep 19 07:42:10 rakete kernel:  txg_wait_open+0xb0/0x100 [zfs]
Sep 19 07:42:10 rakete kernel:  dmu_tx_wait+0x384/0x390 [zfs]
Sep 19 07:42:10 rakete kernel:  dmu_tx_assign+0x8f/0x490 [zfs]
Sep 19 07:42:10 rakete kernel:  zfs_write+0x403/0xd30 [zfs]
Sep 19 07:42:10 rakete kernel:  ? blk_finish_plug+0x2c/0x40
Sep 19 07:42:10 rakete kernel:  ? __do_page_cache_readahead+0x1f4/0x2a0
Sep 19 07:42:10 rakete kernel:  ? dequeue_entity+0xed/0x4d0
Sep 19 07:42:10 rakete kernel:  ? radix_tree_lookup_slot+0x22/0x50
Sep 19 07:42:10 rakete kernel:  ? copyout+0x29/0x30
Sep 19 07:42:10 rakete kernel:  zpl_write_common_iovec+0x8c/0xe0 [zfs]
Sep 19 07:42:10 rakete kernel:  zpl_iter_write+0xae/0xe0 [zfs]
Sep 19 07:42:10 rakete kernel:  new_sync_write+0xcf/0x120
Sep 19 07:42:10 rakete kernel:  __vfs_write+0x26/0x40
Sep 19 07:42:10 rakete kernel:  vfs_write+0xb1/0x1a0
Sep 19 07:42:10 rakete kernel:  SyS_write+0x55/0xc0
Sep 19 07:42:10 rakete kernel:  entry_SYSCALL_64_fastpath+0x1a/0xa5
Sep 19 07:42:10 rakete kernel: RIP: 0033:0x7f9e60ef8844
Sep 19 07:42:10 rakete kernel: RSP: 002b:00007ffd0d5024d8 EFLAGS: 00000246 ORIG_RAX: 0000000000000001
Sep 19 07:42:10 rakete kernel: RAX: ffffffffffffffda RBX: 00005618ffa7b160 RCX: 00007f9e60ef8844
Sep 19 07:42:10 rakete kernel: RDX: 0000000000100000 RSI: 00007f9e61295000 RDI: 0000000000000001
Sep 19 07:42:10 rakete kernel: RBP: 0000000000100000 R08: 00007f9e61398540 R09: 00007ffd0d50249f
Sep 19 07:42:10 rakete kernel: R10: 00007f9e60f85940 R11: 0000000000000246 R12: 0000000000100000
Sep 19 07:42:10 rakete kernel: R13: 0000000000000000 R14: 0000000000000000 R15: 00007f9e61295000
```

Only a reboot helps. the watchdog is letting me wait 3 min before it enters successfully final "shutdown target". Even my internal SATA raidz which holds all my data is unmounted properly. So that is good. Seems like zfs is not completely gone. But unfortunately I have a side effect with my /home (ext4) which can not be unmounted. 

In summary:
1) I think zfs should not crash like this with back trace etc.
2) The side effect with my /home is weird. Any idea why that happens?

Kind Regards
Matthias

**Comment by mabod** [comment]
This issue is obsolete from my point t of view. Closing it.
